### PR TITLE
Introduce Classloader safe implementation of SchemaRegistryClient

### LINKED ISSUE #6902: Kafka connector: ClassCastException while reading data from schema registry table
Getting following exception while querying schema registry table using Kafka connector.
``` 
trino:default> select firstname from "fake-user";
Query 20210215_140138_00007_bfjxk, FAILED, 1 node
http://localhost:8080/ui/query.html?20210215_140138_00007_bfjxk
Splits: 17 total, 0 done (0.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 9% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 0B
0.26 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20210215_140138_00007_bfjxk failed: java.lang.LinkageError: ClassCastException: attempting to castjar:file:/usr/lib/trino/lib/javax.ws.rs-api-2.1.jar!/javax/ws/rs/ext/RuntimeDelegate.class to jar:file:/usr/lib/trino/plugin/kafka/javax.ws.rs-api-2.1.jar!/javax/ws/rs/ext/RuntimeDelegate.class
com.google.common.util.concurrent.ExecutionError: java.lang.LinkageError: ClassCastException: attempting to castjar:file:/usr/lib/trino/lib/javax.ws.rs-api-2.1.jar!/javax/ws/rs/ext/RuntimeDelegate.class to jar:file:/usr/lib/trino/plugin/kafka/javax.ws.rs-api-2.1.jar!/javax/ws/rs/ext/RuntimeDelegate.class
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2049)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3951)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3974)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4935)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4941)
	at io.trino.plugin.kafka.schema.confluent.ConfluentAvroReaderSupplier.get(ConfluentAvroReaderSupplier.java:69)
	at io.trino.decoder.avro.AvroBytesDeserializer.deserialize(AvroBytesDeserializer.java:45)
	at io.trino.decoder.avro.GenericRecordRowDecoder.decodeRow(GenericRecordRowDecoder.java:49)
	at io.trino.decoder.RowDecoder.decodeRow(RowDecoder.java:34)
	at io.trino.plugin.kafka.KafkaRecordSet$KafkaRecordCursor.nextRow(KafkaRecordSet.java:188)
	at io.trino.plugin.kafka.KafkaRecordSet$KafkaRecordCursor.advanceNextPosition(KafkaRecordSet.java:161)
	at io.trino.plugin.kafka.KafkaRecordSet$KafkaRecordCursor.advanceNextPosition(KafkaRecordSet.java:158)
	at io.trino.spi.connector.RecordPageSource.getNextPage(RecordPageSource.java:88)
	at io.trino.operator.TableScanOperator.getOutput(TableScanOperator.java:301)
	at io.trino.operator.Driver.processInternal(Driver.java:387)
	at io.trino.operator.Driver.lambda$processFor$9(Driver.java:291)
	at io.trino.operator.Driver.tryWithLock(Driver.java:683)
	at io.trino.operator.Driver.processFor(Driver.java:284)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1076)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.trino.$gen.Trino_352____20210215_134549_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.LinkageError: ClassCastException: attempting to castjar:file:/usr/lib/trino/lib/javax.ws.rs-api-2.1.jar!/javax/ws/rs/ext/RuntimeDelegate.class to jar:file:/usr/lib/trino/plugin/kafka/javax.ws.rs-api-2.1.jar!/javax/ws/rs/ext/RuntimeDelegate.class
	at javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:149)
	at javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:121)
	at javax.ws.rs.core.UriBuilder.newInstance(UriBuilder.java:96)
	at javax.ws.rs.core.UriBuilder.fromPath(UriBuilder.java:149)
	at io.confluent.kafka.schemaregistry.client.rest.RestService.getId(RestService.java:656)
	at io.confluent.kafka.schemaregistry.client.rest.RestService.getId(RestService.java:642)
	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaByIdFromRegistry(CachedSchemaRegistryClient.java:217)
	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaBySubjectAndId(CachedSchemaRegistryClient.java:291)
	at io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient.getSchemaById(CachedSchemaRegistryClient.java:276)
	at io.confluent.kafka.schemaregistry.client.SchemaRegistryClient.getById(SchemaRegistryClient.java:64)
	at io.trino.plugin.kafka.schema.confluent.ConfluentAvroReaderSupplier.lookupReader(ConfluentAvroReaderSupplier.java:56)
	at com.google.common.cache.CacheLoader$FunctionToCacheLoader.load(CacheLoader.java:165)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045)
	... 24 more

trino:default> 
```
Version used:
[Trino Docker Image ](https://hub.docker.com/layers/trinodb/trino/latest/images/sha256-95f6fc9934802ea8fa66a1716795220cd5d6261d5b81b27ab05f3ca8fdd333b6?context=explore)
[Schema Registry Image](https://hub.docker.com/layers/confluentinc/cp-schema-registry/5.5.2/images/sha256-2becf923fcdaa2ed28770688661a1757ad06a44ac6aa7d5c9cb83298e8e2c6c0?context=explore)

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #6902. 

Overrides #6915

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public void extendEnvironment(Environment.Builder builder)
     {
```
Comment (kokosing) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java:72]: Why `standardMultinode` is not used here?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public void extendEnvironment(Environment.Builder builder)
...
         builder.configureContainer(COORDINATOR, container -> container
                 .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-master-jvm.config")), CONTAINER_PRESTO_JVM_CONFIG)
-                .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-master-config.properties")), CONTAINER_PRESTO_CONFIG_PROPERTIES)
+                .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/standard/multinode-master-config.properties")), CONTAINER_PRESTO_CONFIG_PROPERTIES)
```
Comment (kokosing) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java:75]: this file should be under `conf/environment/standard-multinode/master-config.properties

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ public void extendEnvironment(Environment.Builder builder)
     {
         builder.configureContainer(COORDINATOR, container -> container
                 .withCopyFileToContainer(forHostPath(dockerFiles.getDockerFilesHostPath("conf/environment/multinode/multinode-master-jvm.config")), CONTAINER_PRESTO_JVM_CONFIG)
```
Comment (kokosing) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java:74]: while you are here can use `io.trino.tests.product.launcher.docker.DockerFiles.ResourceProvider` here. It is cleaner and safer.

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestClassloaderSafeSchemaRegistryClient.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package io.trino.plugin.kafka.schema.confluent;
```
Comment (kokosing) [inline plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestClassloaderSafeSchemaRegistryClient.java:1]: copyright is missing

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeKafka.java [hunk 5]
```diff
@@ -17,48 +17,57 @@
...
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.COORDINATOR;
 import static io.trino.tests.product.launcher.env.EnvironmentContainers.TESTS;
+import static io.trino.tests.product.launcher.env.EnvironmentContainers.WORKER;
 import static io.trino.tests.product.launcher.env.common.Standard.CONTAINER_PRESTO_ETC;
 import static java.util.Objects.requireNonNull;
 import static org.testcontainers.utility.MountableFile.forHostPath;
 
 @TestsEnvironment
-public final class SinglenodeKafka
+public final class MultinodeKafka
```
Comment (skrzypo987) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeKafka.java:37]: Don't we want to test both single and multinode?

Comment (kokosing) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeKafka.java:37]: Mutlinode covers singlenode.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java (hunk 1 above)
Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java:72]: `standardMultinode` creates a single worker - while `MulitnodeHiveCaching` requires creating more than one workers

Comment (kokosing) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/MultinodeHiveCaching.java:74]: Maybe we should make `standardMultinode` to be parametric. Anyway, out of the scope of this PR.

**Review by kokosing — APPROVED** [review]

**Comment by findepi** [comment]
@wendigo can you please review Launcher changes?

**Comment by electrum** [comment]
Nice, this seems like a cleaner and more performant approach.

File: plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestClasslLoaderSafeSchemaRegistryClient.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka.schema.confluent;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import org.testng.annotations.Test;
+
+import static io.trino.spi.testing.InterfaceTestUtils.assertAllMethodsOverridden;
+
+public class TestClasslLoaderSafeSchemaRegistryClient
```
Comment (wendigo) [inline plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/schema/confluent/TestClasslLoaderSafeSchemaRegistryClient.java:21]:
```suggestion
public class TestClassLoaderSafeSchemaRegistryClient
```

File: plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ClassloaderSafeSchemaRegistryClient.java [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ClassloaderSafeSchemaRegistryClient
```
Comment (wendigo) [inline plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ClassloaderSafeSchemaRegistryClient.java:33]:
```suggestion
public class ClassLoaderSafeSchemaRegistryClient
```

**Review by wendigo — APPROVED** [review]

**Review by skrzypo987 — APPROVED** [review]

**Review by s2lomon — APPROVED** [review]
It looks good to me

**Pull request merged**